activerecord/test/cases/persistence_test.rb

```diff
@@ -43,31 +43,42 @@ def test_populates_non_primary_key_autoincremented_column_for_a_cpk_model
     assert_not_nil order_id
   end
 
-  def test_fills_auto_populated_columns_on_creation
-    record_with_defaults = Default.create
-    assert_not_nil record_with_defaults.id
-    assert_equal "Ruby on Rails", record_with_defaults.ruby_on_rails
-
-    if current_adapter?(:PostgreSQLAdapter) && ActiveRecord::Base.connection.supports_virtual_columns?
-      assert_not_nil record_with_defaults.virtual_stored_number
-    end
-
-    assert_not_nil record_with_defaults.random_number
-    assert_not_nil record_with_defaults.modified_date
-    assert_not_nil record_with_defaults.modified_date_function
-    assert_not_nil record_with_defaults.modified_time
-    assert_not_nil record_with_defaults.modified_time_without_precision
-    assert_not_nil record_with_defaults.modified_time_function
+  if current_adapter?(:PostgreSQLAdapter, :SQLite3Adapter, :Mysql2Adapter, :TrilogyAdapter)
+    def test_fills_auto_populated_columns_on_creation
+      record = Default.create
+      assert_not_nil record.id
 
-    if current_adapter?(:PostgreSQLAdapter) && ActiveRecord::Base.connection.supports_identity_columns?
-      klass = Class.new(ActiveRecord::Base) do
-        self.table_name = "postgresql_identity_table"
+      if current_adapter?(:PostgreSQLAdapter, :SQLite3Adapter)
```

Comment:
I'm finding it difficult to follow this test, since we're asserting different things for different adapters 😅 Maybe we could pull the assertions that are adapter-specific out into separate tests, one per adapter?

Reply:
Agreed, extracted as separate test cases.